eth/contracts/libraries/LibArtifact.sol

```diff
@@ -0,0 +1,351 @@
+            artifactType = ArtifactType.PhotoidCannon;
+        } else if (lastByteOfSeed < 216) {
+            artifactType = ArtifactType.BloomFilter;
+        } else if (lastByteOfSeed < 231) {
+            artifactType = ArtifactType.BlackDomain;
+        } else {
+            if (biome == Biome.Ice) {
+                artifactType = ArtifactType.PlanetaryShield;
+            } else if (biome == Biome.Lava) {
+                artifactType = ArtifactType.PhotoidCannon;
+            } else if (biome == Biome.Wasteland) {
+                artifactType = ArtifactType.BloomFilter;
+            } else if (biome == Biome.Corrupted) {
+                artifactType = ArtifactType.BlackDomain;
+            } else {
+                artifactType = ArtifactType.Wormhole;
+            }
+            artifactType = ArtifactType.PhotoidCannon;
+        }
+
+        uint256 bonus = 0;
+        if (secondLastByteOfSeed < 4) {
+            bonus = 2;
+        } else if (secondLastByteOfSeed < 16) {
+            bonus = 1;
+        }
+
+        return (artifactType, bonus);
+    }
+
+    // planets can have multiple artifacts on them. this function updates all the
+    // internal contract book-keeping to reflect that the given artifact was
+    // put on. note that this function does not transfer the artifact.
+    function putArtifactOnPlanet(uint256 locationId, uint256 artifactId) internal {
+        gs().planetArtifacts[locationId].push(artifactId);
+    }
+
+    /**
+     * Remove artifactId from planet with locationId if artifactId exists AND is not active.
+     */
+    function takeArtifactOffPlanet(uint256 locationId, uint256 artifactId) internal {
+        uint256 artifactsOnThisPlanet = gs().planetArtifacts[locationId].length;
+
+        bool hadTheArtifact = false;
+
+        for (uint256 i = 0; i < artifactsOnThisPlanet; i++) {
+            if (gs().planetArtifacts[locationId][i] == artifactId) {
+                require(
+                    !isActivated(locationId, artifactId),
+                    "you cannot take an activated artifact off a planet"
+                );
+
+                gs().planetArtifacts[locationId][i] = gs().planetArtifacts[locationId][
+                    artifactsOnThisPlanet - 1
+                ];
+
+                hadTheArtifact = true;
+                break;
+            }
+        }
+
+        require(hadTheArtifact, "this artifact was not present on this planet");
+        gs().planetArtifacts[locationId].pop();
+    }
+
+    // if the given planet has an activated artifact on it, then return the artifact
+    // otherwise, return a 'null artifact'
+    function getActiveArtifact(uint256 locationId) internal view returns (Artifact memory) {
+        if (hasActiveArtifact(locationId)) {
+            uint256 artifactId = gs().planetActiveArtifact[locationId];
+            return LibArtifact.decode(artifactId);
+        } else {
+            return LibArtifact._nullArtifactProperties();
```

Comment:
This is used in `_checkWormhole` and `_checkPhotoid` in `DFMoveFacet.sol`. We need a way to see if an artifact is the correct type. This function combines two bits of logic: 1) See if active artifact exists. 2) Return activate artifact. 

I think if we split it into two separate functions and have `getActiveArtifact` revert if there is no activate artifact, it would be cleaner. I'll do that.

Reply:
6471c89950df7c09ee764f61232372d7c03a4cea